### PR TITLE
Update metadata documentation

### PR DESCRIPTION
**Pull request by pabloarosado**
Update metadata documentation in schemas.

### CODE REVIEW

File: schemas/dataset-schema.json [hunk 1]
```diff
@@ -217,23 +217,38 @@
       "properties": {
         "title": {
           "type": "string",
-          "title": "Title of this ETL dataset",
-          "description": "Title of the dataset (mostly for internal purposes, or for users of our data catalog) which is a one-line description of the dataset (follow the guidelines of the origin's `data_product_title`). NOTE: Dataset titles should be propagated automatically from snapshots. By default, the title of the dataset will be the title of the containing table. But, if there are multiple tables, or if the user wants to manually edit the dataset title, it can be done by editing `dataset.title`.",
-          "requirement_level": "required (often automatic)"
+          "title": "Title of the ETL dataset",
+          "description": "Title of the dataset (mostly for internal purposes, or for users of our data catalog) which is a one-line description of the dataset. NOTE: Dataset titles should be propagated automatically from snapshots. By default, the title of the dataset will be the title of the containing table. But, if there are multiple tables, or if the user wants to manually edit the dataset title, it can be done by editing `dataset.title`.",
```
Comment (lucasrodes) [inline schemas/dataset-schema.json:221]: > to manually edit the dataset title, it can be done by editing `dataset.title`.

I find it strange to refer to the field itself with the explicit name. I'd go for

> to manually edit the dataset title, it can be done by editing _this field_.

Comment (lucasrodes) [reply schemas/dataset-schema.json:221]: Also, the description seems long enough that it would benefit from some line breaks?

> Title of the dataset (mostly for internal purposes, or for users of our data catalog) which is a one-line description of the dataset.\n\n**Note:** Dataset titles should be propagated automatically from snapshots. By default, the title of the dataset will be the title of the containing table. But, if there are multiple tables, or if the user wants to manually edit the dataset title, it can be done by editing this field.

File: schemas/dataset-schema.json [hunk 2]
```diff
@@ -217,23 +217,38 @@
...
-          "description": "Description of the dataset (mostly for internal purposes, or for users of our data catalog) which is a one- (or a few) paragraph description of the dataset (follow the guidelines of the origin's `description`). NOTE: Dataset descriptions should be propagated automatically from snapshots. By default, the description of the dataset will be the description of the containing table. But, if there are multiple tables, or if the user wants to manually edit the dataset description, it can be done by editing `dataset.description`.",
-          "requirement_level": "recommended (often automatic)"
+          "title": "Description of the ETL dataset",
+          "description": "Description of the dataset (mostly for internal purposes, or for users of our data catalog) which is a one- (or a few) paragraph description of the content of the tables. NOTE: Dataset descriptions should be propagated automatically from snapshots. By default, the description of the dataset will be the description of the containing table. But, if there are multiple tables, or if the user wants to manually edit the dataset description, it can be done by editing `dataset.description`.",
```
Comment (lucasrodes) [inline schemas/dataset-schema.json:233]: Idem comments as in `dataset.title`. Would probably add a line break and make it `dataset.description` → _this field_.

File: schemas/dataset-schema.json [hunk 3]
```diff
@@ -217,23 +217,38 @@
...
+          "requirement_level": "recommended (often automatic)",
+          "guidelines": [
+            ["Must start with a capital letter."],
+            ["Must end with a period."],
+            ["Must not mention other metadata fields like `producer`.", {"type": "exceptions", "value": ["Other metadata fields are crucial in the description of the data product."]}],
```
Comment (lucasrodes) [inline schemas/dataset-schema.json:238]: I would phrase it like

"Must not mention other metadata fields (e.g. `producer` or ...)." (add another metadata field)

and then

"The other metadata fields are crucial in describing the data product."

File: schemas/dataset-schema.json [hunk 4]
```diff
@@ -269,15 +284,28 @@
         "properties": {
           "title": {
             "type": "string",
-            "title": "Title of this ETL table",
-            "description": "Title of the table (mostly for internal purposes, or for users of our data catalog) which is a few words description of the table (follow the guidelines as origin's `title`). NOTE: Table titles should be propagated automatically from snapshots (from origin's `title`). But, if there are multiple tables, or if the user wants to manually edit the title, it can be done by editing `table.title`.",
-            "requirement_level": "required (often automatic)"
+            "title": "Title of the ETL table",
+            "description": "Title of the table (mostly for internal purposes, or for users of our data catalog) which is a few words description of the table. NOTE: Table titles should be propagated automatically from snapshots (from origin's `title`). But, if there are multiple tables, or if the user wants to manually edit the title, it can be done by editing `table.title`.",
```
Comment (lucasrodes) [inline schemas/dataset-schema.json:288]: same comments as in `dataset.title` and `dataset.description` (e.g. breakline and referring to the field as _this field_).

File: schemas/dataset-schema.json [hunk 5]
```diff
@@ -269,15 +284,28 @@
...
+            ]
           },
           "description": {
             "type": "string",
             "title": "Description of this ETL table",
-            "description": "Description of the table (mostly for internal purposes, or for users of our data catalog) which is a one- (or a few) paragraph description of the table (follow the guidelines of origin's `description`). NOTE: Table descriptions should be propagated automatically from snapshots (from origin's `description`). But, if there are multiple tables, or if the user wants to manually edit the description, it can be done by editing `table.description`.",
-            "requirement_level": "recommended (often automatic)"
+            "description": "Description of the table (mostly for internal purposes, or for users of our data catalog) which is a one- (or a few) paragraph description of the table. NOTE: Table descriptions should be propagated automatically from snapshots (from origin's `description`). But, if there are multiple tables, or if the user wants to manually edit the description, it can be done by editing `table.description`.",
```
Comment (lucasrodes) [inline schemas/dataset-schema.json:300]: same comments as in `tables.title` (e.g. breakline and referring to the field as this field).

File: schemas/dataset-schema.json [hunk 6]
```diff
@@ -269,15 +284,28 @@
...
+            "requirement_level": "recommended (often automatic)",
+            "guidelines": [
+              ["Must start with a capital letter."],
+              ["Must end with a period."],
+              ["Must not mention other metadata fields like `producer`.", {"type": "exceptions", "value": ["Other metadata fields are crucial in the description of the table."]}],
```
Comment (lucasrodes) [inline schemas/dataset-schema.json:305]: I would phrase it like

"Must not mention other metadata fields (e.g. `producer` or ...)." (add another metadata field)

and then

"The other metadata fields are crucial in describing the data product."

File: schemas/dataset-schema.json [hunk 7]
```diff
@@ -442,7 +470,7 @@
                 "display": { "$ref": "definitions.json#/display" },
                 "presentation_license": {
                   "type": "object",
-                  "description": "Custom license for the indicator. This is usally not needed and the value from `license` is used.",
+                  "description": "License to display for the indicator, overriding `license`.",
```
Comment (lucasrodes) [inline schemas/dataset-schema.json:473]: I am not sure what's the best way to write the field `license`. It would be something like `tables[].variables[].license`? Either write it as that or `indicator.license`?

File: schemas/dataset-schema.json [hunk 8]
```diff
@@ -462,8 +490,16 @@
                   "properties": {
                     "title_public": {
                       "type": "string",
-                      "description": "Indicator title to be shown in data pages (follow the guidelines of indicator's `title`). NOTE: This may be needed, for example, when the indicator comes from a big dataset where titles can't easily be curated manually. For those cases, `title_public` will override the indicator's `title`.",
-                      "requirement_level": "optional"
+                      "description": "Indicator title to be shown in data pages. NOTE: This may be needed, for example, when the indicator comes from a big dataset where titles can't easily be curated manually. For those cases, `title_public` will override the indicator's `title`.",
```
Comment (lucasrodes) [inline schemas/dataset-schema.json:493]: Same as in `dataset.title` (linebreaks)

File: schemas/dataset-schema.json [hunk 9]
```diff
@@ -488,22 +524,29 @@
                     "attribution": {
                       "type": "string",
                       "title": "Indicator's attribution",
-                      "description": "",
+                      "description": "Citation of the indicator's origins, to be used when the automatic format `producer1 (year1); producer2 (year2)` needs to be overridden.",
```
Comment (lucasrodes) [inline schemas/dataset-schema.json:527]: I'd rephrase it as 

> Citation of the indicator's origins. Use it only when you want to overwrite the automatic format `producer1 (year1); producer2 (year2)`.

but not really important

File: schemas/definitions.json [hunk 10]
```diff
@@ -269,43 +272,52 @@
         "guidelines": [
           ["Must start with a capital letter."],
           ["Must end with a period."],
-          ["Should describe the data product, not the snapshot (i.e. the subset of data we extract from the data product)."]
+          ["Must not mention other metadata fields like `producer` or `version_producer`.", {"type": "exceptions", "value": ["Other metadata fields are crucial in the description of the data product."]}],
```
Comment (lucasrodes) [inline schemas/definitions.json:275]: would modify

"Other metadata fields are crucial in the description of the data product."

to

"These other metadata fields are crucial in the description of the data product."

File: schemas/definitions.json [hunk 11]
```diff
@@ -269,43 +272,52 @@
...
           "Neutron star mergers"
         ],
         "examples_bad": [
           [
+            "Global Carbon Budget",
             "Neutron star mergers (NASA, 2023)",
             "Data on neutron star mergers",
             "Neutron star mergers dataset"
           ]
         ],
```
Comment (lucasrodes) [inline schemas/definitions.json:297]: To correctly render the list of examples in the docs or wizard we should either:

- not have any `examples` or `examples_bad`
- just have `examples`
- have the same number of `examples` and `examples_bad`.

Here I'd do sth like


```json
"examples": [
  "Global Carbon Budget - Fossil fuels",
  "Neutron star mergers"
],
"examples_bad": [
  [
    "Global Carbon Budget",
  ],
  [
    "Neutron star mergers (NASA, 2023)",
    "Data on neutron star mergers",
    "Neutron star mergers dataset"
  ]
],
```


(see how it is done for `attribution_short`)

File: schemas/definitions.json [hunk 12]
```diff
@@ -315,28 +327,30 @@
...
         "guidelines": [
           ["Must start with a capital letter.", {"type": "exceptions", "value": ["The name of the institution or the author must be spelled with small letter, e.g. `van Haasteren`."]}],
           ["Must not end with a period."],
           ["Must end with the year of `date_published` in parenthesis."],
           ["Must not include any semicolon `;`."],
-          ["Should only be used if the automatic attribution format `producer (year)`is incomplete. For example, when the dataset title is well known and should be cited along with the producer, or when the original dataset version should also be mentioned."],
-          ["Should be constructed using placeholders, e.g. `{title}, {producer} ({year})`."]
+          ["Should only be used if the automatic attribution format `producer (year)` is considered uninformative. For example, when the title of the data product is well known and should be cited along with the producer, or when the original version of the data product should also be mentioned."],
+          ["Should follow the format `{producer} - {title} {version_producer} ({year})`."],
```
Comment (lucasrodes) [inline schemas/definitions.json:345]: I am a bit confused with this one. Thought `attribution` was free, but from this guideline it seems that there is a specific style we should follow?

**Review by lucasrodes — COMMENTED** [review]
Thanks, Pablo! LGTM in general, added minor comments.

File: schemas/dataset-schema.json (hunk 7 above)
Comment (pabloarosado) [reply schemas/dataset-schema.json:473]: I didn't even know about this field called "presentation_license". Clearly it should be under "presentation". Everything around licenses is still unclear. We may have to rephrase these things. Let's leave it for not to not block other things.

File: schemas/definitions.json (hunk 11 above)
Comment (pabloarosado) [reply schemas/definitions.json:297]: Right, I may have other examples that don't fulfil this rule.

File: schemas/definitions.json (hunk 12 above)
Comment (pabloarosado) [reply schemas/definitions.json:345]: I've rephrased it. This is a "should" not a "must". My point here is that, if you are simply including the data product, then that should be the preferred way to do so. If you want to do something completely different, you are free.
